pinot-common/src/main/java/org/apache/pinot/common/utils/RoundRobinURIProvider.java

```diff
@@ -63,4 +78,33 @@ public URI next() {
     _index = (_index + 1) % _uris.length;
     return result;
   }
+
+  public URI[] resolveHostToIPAddresses(URI originalUri)
```

Comment:
I think this and the following method shouldn't be public, since we are anyways taking these args as input. To be more precise: if we create an object of `RoundRobinURIProvider` with some URIs, then we shouldn't allow callers to bypass those URIs (they should simply call `#next`)

Reply:
Ah my bad missed this! Updated!